quil-rs/src/expression/mod.rs

```diff
@@ -990,27 +1003,38 @@ mod tests {
             prop_assert!(p.is_ok());
             let p = p.unwrap();
             let simple_p = p.clone().into_simplified();
-            prop_assert_eq!(
-                simple_p.clone(),
-                simple_e.clone(),
-                "Simplified expressions should be equal:\nparenthesized {p} ({p:?}) extracted from {s} simplified to {simple_p}\nvs original {e} ({e:?}) simplified to {simple_e}",
-                p=p.to_quil_or_debug(),
-                s=s,
-                e=e.to_quil_or_debug(),
-                simple_p=simple_p.to_quil_or_debug(),
-                simple_e=simple_e.to_quil_or_debug()
-            );
+
+            if !is_nan(&simple_p) && !is_nan(&simple_e) {
+                prop_assert_eq!(
+                    simple_p.clone(),
+                    simple_e.clone(),
+                    "Simplified expressions should be equal:\nparenthesized {p} ({p:?}) extracted from {s} simplified to {simple_p}\nvs original {e} ({e:?}) simplified to {simple_e}",
+                    p=p.to_quil_or_debug(),
+                    s=s,
+                    e=e.to_quil_or_debug(),
+                    simple_p=simple_p.to_quil_or_debug(),
+                    simple_e=simple_e.to_quil_or_debug()
+                );
+            }
         }
 
     }
 
+    fn is_nan(expression: &Expression) -> bool {
+        match expression {
+            Expression::Number(value) => value.is_nan(),
+            _ => false,
+        }
```

Comment:
What about the case where an `Expression` has `NaN` nested somewhere inside it?

Reply:
Uff, good point. Seems like we should re-write `impl PartialEq for Expression`. Specifically,

```rs
(Self::Number(left), Self::Number(right)) => left == right
```

should explicitly check for NaN?

Reply:
That seems fine to me

Reply:
I just filtered `/ 0` out of any prop test generated expressions. This won't affect any non-test code.